GitCommands/Settings/AppSettings.cs

```diff
@@ -588,10 +588,16 @@ public static bool ShowAuthorAvatarInCommitInfo
 
         public static AvatarProvider AvatarProvider
         {
-            get => GetEnumViaString("Appearance.AvatarProvider", AvatarProvider.Default);
+            get => GetEnumViaString("Appearance.AvatarProvider", AvatarProvider.None);
```

Comment:
In fact, the Initials is the default value of `AvatarFallbackType` settings. So yes, initials will end up to be the default avatar image on a fresh install. It has the advantage to be really fast, 100% local, much more recognizable than other fallback and do not expire...

Reply:
Code in c# a way to get composite color so no matter the background color a text color will show. 

To get a composite color that will contrast with any given background color, you can use the following C# code:

scssCopy code

Color GetContrastColor(Color backgroundColor) { // Calculate the luminance of the background color double luminance = (0.2126 * backgroundColor.R + 0.7152 * backgroundColor.G + 0.0722 * backgroundColor.B) / 255; // Use a threshold value to determine whether to use black or white as the foreground color Color foregroundColor = (luminance > 0.5) ? Color.Black : Color.White; return foregroundColor; } 

This method takes a Color object representing the background color and returns a Color object representing the foreground color that will provide the best contrast with the background.

The method works by calculating the luminance of the background color using the standard formula for converting RGB values to grayscale. If the luminance is greater than 0.5, it means the background color is relatively light, so the method returns black as the foreground color. If the luminance is less than or equal to 0.5, the background color is relatively dark, so the method returns white as the foreground color.

You can use this method in your code by calling GetContrastColor() with the background color you want to contrast with. For example:

scssCopy code

Color backgroundColor = Color.FromArgb(255, 128, 0); // orange Color foregroundColor = GetContrastColor(backgroundColor); Console.WriteLine($"Foreground color: {foregroundColor}"); 

This code will output "Foreground color: Color [A=255, R=255, G=255, B=255]" because white is the best foreground color to use with an orange background

Reply:
git-range-diff was very helpful in this review....

@vbjay Thanks for the response, quoted in the other comment.

Reply:
No prob. Why I sent it was exactly for the idea of when theming works. The
dark theme can have a gradient of light color backgrounds to show a good
contrast on top of a dark background. Along with text color being good on
said background. Then flip the idea on light backgrounds.

I used chatgpt and thought it would be cool to just copy the original
convo.

Github email handler could be better  If replying to a review thread email, should put reply in that thread.